build-scripts/multidirectory.sh

```diff
@@ -0,0 +1,57 @@
+#!/bin/sh
+# Creates/removes/updates branches according the `jobs/` directory on the
+# master branch.  Local files and branches will be modified, so this script
+# should be executed on a clean, dedicated clone.
+set -eu
+
+set_difference() {
+    printf '%s\n%s\n%s\n' "$1" "$1" "$2" | sort | uniq -u
+}
+
+list_branches() {
+    git branch --list --remotes 'origin/*' \
+        | awk '$1 ~ /^origin\/.*\// {print gensub("^origin/", "", 1, $1)}'
+}
+
+list_jobs() {
+    [ -e jobs/ ] || return 0
```

Comment:
`cd jobs/ && find …` is mainly to get the output in the correct form (i.e.
without having to remove the `jobs/` prefix).  You're right about `dirname`
though, I completely forgot about it, but I knew there *had* to be a better way
to do this.